mixins/notify/index.js

```diff
@@ -0,0 +1,30 @@
+export default {
+  methods: {
+    /**
+     * Toast notification indicating success
+     * @param {String} message
+     */
+    notifySuccess: function(message) {
+      this.$message({
+        message: message,
+        showClose: true,
+        iconClass: 'el-icon-circle-check',
+        customClass: 'el-message--success',
+        duration: 5000
+      })
+    },
+    /**
+     * Toast notification indicating failure
+     * @param {String} message
+     */
+    notifyFail: function(message) {
+      this.$message({
+        message: message,
+        showClose: true,
+        iconClass: 'el-icon-circle-close',
+        customClass: 'el-message--error',
+        duration: 5000
+      })
+    }
+  }
+}
```

Comment:
I'm just really not a fan of the mixins with Vue 2 because it becomes so hard to track down where everything on the component `this` instance is coming from.

Answer me this...how does `$message` even get injected into the class?  I'm assuming its somewhere near `index.js` where all the other global stuff is injected into every `Vue` instance, but I wouldn't know without digging...

One way to turn this into a pure function instead of a mixin is just to write a function that takes a string and returns the object that you want to send to the `$message` function:

```
export const failMessage = message => ({
  message,
  showClose: true,
  iconClass: 'el-icon-circle-close',
  customClass: 'el-message--error',
  duration: 5000
)}
```

and now in your component
```
import { successMessage, failMessage } from '...wherever/this/file/is'

...rest of your script

handleCitationCopy: function() {
  ...
  this.$message(failMessage('Failed to copy citation.'))
}
```

Reply:
`$message` is an Element component that gets added on to the [Vue prototype if imported](https://element.eleme.io/?#/en-US/component/message#global-method). Changing it to something like that pure function would work well.

Reply:
Makes sense, really good explanation, I agree.